### ISSUE TITLE
`doom-apply-ansi-color-to-compilation-buffer-h` does not work when compiling Haskell code (GHC)

### ISSUE THREAD

**Issue by dschrempf** [hlissner/doom-emacs#6222]
### What did you expect to happen?

Compilation finishes without error messages.

### What actually happened?

Various errors are reported by `doom-apply-ansi-color-to-compilation-buffer-h` which fails to colorize the compilation output of GHC.

### Describe your attempts to resolve the issue

I added
```
;; Compile.
(after! compile
  ;; The ANSI color hook leads to errors when compiling Haskell code.
  (remove-hook 'compilation-filter-hook #'doom-apply-ansi-color-to-compilation-buffer-h))
```
to my configuration. Then, compilation with GHC works perfect (with colors). However, compilation with `gcc` is now ugly.

Is there an option to turn this hook off for specific modes only?

### Steps to reproduce

Compile Haskell code or C code.

### System Information

https://pastebin.com/bpxTuqSc

**Comment by hlissner** [comment]
Does `(setq ansi-color-for-compilation-mode 'filter)` fix this?

**Comment by dschrempf** [comment]
Yes, thank you `(setq ansi-color-for-compilation-mode 'filter)` fixes this for both, GHC and GCC compilation buffers. Should we add this to the Doom configuration? Otherwise, we should close this issue! Thanks.

**Comment by dschrempf** [comment]
Sorry I spoke too soon. The GHC compilation is failing again, with both options. Somehow this error does not always appear though...

**Comment by dschrempf** [comment]
I get a lot of:

```
error in process filter: ansi-color-apply-on-region: Invalid search bound (wrong side of point)
error in process filter: Invalid search bound (wrong side of point)
```

Using

```elisp
(after! compile
  (remove-hook 'compilation-filter-hook #'doom-apply-ansi-color-to-compilation-buffer-h)
  (add-hook 'compilation-filter-hook #'ansi-color-compilation-filter))
```
as you suggested above, does not help.

**Comment by dschrempf** [comment]
BTW, I have to remove more things from the compilation filter hook now:

```
;; Compile.
(after! compile
  ;; The ANSI color hook leads to errors when compiling Haskell code.
  (remove-hook 'compilation-filter-hook #'doom-apply-ansi-color-to-compilation-buffer-h)
  (remove-hook 'compilation-filter-hook #'ansi-color-compilation-filter))
```

Otherwise, I get errors.

**Comment by hlissner** [comment]
`doom-apply-ansi-color-to-compilation-buffer-h` was removed in 2f7171bf8a5f76c1fbb3b9c91921be270b09c0b5, and replaced with `ansi-color-compilation-filter`, so you only need to remove that. I haven't been able to reproduce this yet, however.

**Comment by dschrempf** [comment]
Maybe one needs actual ansi color signs for this to pop up. The only colored word I see in my Haskell compilation buffer is the "Warning" keyword when compiling with cabal (I use Nix, and do not populate `~/.cabal`):

```
Warning: The package list for 'hackage.haskell.org' does not exist. Run 'cabal
update' to download it.
```

**Comment by xuning97** [comment]
I encountered a related issue, that ansi color code is part of the file name when I try to jump to the compilation error location.
<img width="424" alt="image" src="https://github.com/doomemacs/doomemacs/assets/1954442/c7673af4-acee-4aec-bc32-0cd6a575f7a9">

**Comment by LemonBreezes** [comment]
> I encountered a related issue, that ansi color code is part of the file name when I try to jump to the compilation error location. <img alt="image" width="424" src="https://private-user-images.githubusercontent.com/1954442/265659914-c7673af4-acee-4aec-bc32-0cd6a575f7a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTc4OTgyOTgsIm5iZiI6MTcxNzg5Nzk5OCwicGF0aCI6Ii8xOTU0NDQyLzI2NTY1OTkxNC1jNzY3M2FmNC1hY2VlLTRhZWMtYmMzMi0wY2Q2YTU3NWY3YTkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MDlUMDE1MzE4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTExZGZkZjYzMWViZDdjNTRjMDlkZDlkNzg5ZjY5N2U1YTUxMTg2YzRhM2QzMmIxMzdmOTZlMmU4M2VhMGM1MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.1eNmfMEgm5OMK2vJd9Z7rIn6h6T3L4JUYL0CHPdN-_A">

Try 
```elisp
(after! compile (remove-hook 'compilation-filter-hook #'comint-truncate-buffer))
```
I'm testing this out to see if it fixes the issue.
